src/sources/docker_logs.rs

```diff
@@ -75,26 +77,39 @@ impl Default for DockerLogsConfig {
 }
 
 impl DockerLogsConfig {
-    fn container_name_included<'a>(
+    fn container_name_or_id_included<'a>(
         &self,
         id: &str,
         names: impl IntoIterator<Item = &'a str>,
     ) -> bool {
-        if let Some(include_containers) = &self.include_containers {
-            let id_flag = include_containers
-                .iter()
-                .any(|include| id.starts_with(include));
+        let containers: Vec<String> = names.into_iter().map(|i| i.into()).collect();
 
-            let name_flag = names.into_iter().any(|name| {
-                include_containers
-                    .iter()
-                    .any(|include| name.starts_with(include))
-            });
+        // Containers are included by default
+        let mut included: bool = true;
 
-            id_flag || name_flag
-        } else {
-            true
+        // Container remains included if include_containers is specified
+        // and container name or ID matches the inclusion list
+        if let Some(include_list) = &self.include_containers {
+            included = Self::name_or_id_matches(id, &containers, include_list);
+        }
+
+        // Container gets excluded if exclude_containers is specified
+        // and container name or ID matches the exclusion list
+        if let Some(exclude_list) = &self.exclude_containers {
+            included = !(Self::name_or_id_matches(id, &containers, exclude_list));
         }
```

Comment:
I don't think this logic works, since the latter value of `included` overwrites the former, where it should be anded. I think this could simply be:
```suggestion
        let included = self.include_containers.as_ref().map(|include_list| Self::name_or_id_matches(id, &containers, include_list))
            && !self.exclude_containers.as_ref().map(|exclude_list| Self::name_or_id_matches(id, &containers, exclude_list));
```

Reply:
Missed this before I left my review; this would be a good way to do it too rather than my `match` below.

Reply:
You are right about this. I've opted, however, for @jszwedko's suggestion to match on a `(&self.include_containers, &self.exclude_containers)` tuple, which I think is a teensy bit more elegant.

Reply:
I actually initially went with the match statement but then decided on Bruce's suggestion 😄 Glad we're all aligned here 👍🏼